### PR TITLE
Re-implement cycle detection and syntax checking with new backbone objects (#517)

### LINKED ISSUE #517: Re-implement cycle detection and syntax checking with new backbone objects
We disabled two key algorithmic checks in order to move the backbone project forward. `syntaxCheck` checks to see if the model is well formed (i.e., it checks to see that decomposition relationships do not mix `and` and `or`). `cycleSearch()` determines whether the graph contains a cycle. If a cycle is detected (not allowed by analysis), a pop-up is shown, then the cycle is highlighted and the user is kept in Modeling mode.

Steps:
- [x] Trace through `syntaxCheck()` and `cycleSearch()` in `errorDetection.js` on the `develop` branch to understand how these algorithms work
- [x] Update `$('#analysis-btn').on('click', function () {` in `onFunctions.js` using commented out code at top of `errorDetection.js`
- [x] Update the `syntaxCheck()` function (in `errorDetection.js`) to use the new backbone objects (and all associated helper functions)
- [x] Update the `cycleSearch()` function (in `errorDetection.js`) to use the new backbone objects (and all associated helper functions)
- [x] Test/fix `Clear Cycle Highlighting` button (in `Edit` menu now, but in `Clear` on `develop` branch).

### PR DESCRIPTION
**Pull request by amgrubb**
Fixes #517

### CODE REVIEW

**Comment by i-m-foster** [comment]
I think you accidentally commit app.js. Also, did you mean to change the blooming.jar file

**Review by i-m-foster — DISMISSED** [review]
Besides uncommitting blooming.jar i think it looks good! (as far as I can tell)

File: leaf-ui/js/errorDetection.js [hunk 1]
```diff
@@ -5,26 +5,18 @@
...
-//         switchToAnalysisMode();
-//     } 
-// });
-
- /**
-  * Changes all intentions to their original colors
-  * Note: if this is ever merged with the color-visualization branch, EVO will need to be refreshed here
-  */
+// TODO: Find out if this line is necessary 
+// const { connected } = require("process");
```
Comment (Manarnz) [inline leaf-ui/js/errorDetection.js:9]: Determine that now (?)

File: leaf-ui/js/errorDetection.js [hunk 2]
```diff
@@ -103,30 +95,32 @@ function initColorList() {
...
-
-        if (inputlinks[j].postType != null) {
-            constraint = inputlinks[j].linkType+"|"+inputlinks[j].postType;
-        }else {
-            constraint = inputlinks[j].linkType;
+		if (links[j].attributes.link.attributes.postType != null) {
+			constraint = links[j].attributes.link.attributes.linkType + "|" + 
+			links[j].attributes.link.attributes.postType; 
+        } else {
+			constraint = links[j].attributes.link.attributes.linkType;
```
Comment (Manarnz) [inline leaf-ui/js/errorDetection.js:117]: remove extra space after `;` same in line 115

File: leaf-ui/js/errorDetection.js [hunk 3]
```diff
@@ -312,96 +307,117 @@ function alertSyntaxError(title, message) {
...
     }
 
+	// If errorText is not empty
     if (errorText) {
     	alertSyntaxError('We found invalid link combinations', errorText);
     	return true;
-    }
-    return false;
+	}
+	return false;
```
Comment (Manarnz) [inline leaf-ui/js/errorDetection.js:333]: delete extra space after `;`

File: leaf-ui/js/errorDetection.js [hunk 4]
```diff
@@ -312,96 +307,117 @@ function alertSyntaxError(title, message) {
...
  * @returns {null} otherwise
  */
 function cycleSearch() {
-	var links = getLinks();
+	var links = graph.getLinks();
 	var vertices = getElementList();
-	var isCycle = false;
 
-	//initialize linkMap, a 2D array. 1st index = src nodeID. Subarray at index = dest nodes corresponding to the src
+	// Initialize LinkMap, a HashMap. Index = src ID, value = Array of dest nodes(?)
```
Comment (Manarnz) [inline leaf-ui/js/errorDetection.js:384]: Why the `(?)`?

**Review by Manarnz — DISMISSED** [review]
Great job Angela!!!

**Comment by angelac2023** [comment]
> I think you accidentally commit app.js. Also, did you mean to change the blooming.jar file

The first half has been resolved.

**Review by Domily11 — DISMISSED** [review]
From what I can accurately see, it good?

**Comment by amgrubb** [comment]
Approved!

**Review by YesugenB — APPROVED** [review]

**Review by Manarnz — APPROVED** [review]

**Review by Domily11 — APPROVED** [review]

**Pull request merged**